src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,125 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double SHOULD_GROW_TIMES = 1.5;
+    private static final int STARTING_POSITION = 0;
+    private int size;
+    private Object[] arrayList;
+
+    public ArrayList() {
+        arrayList = new Object[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void add(T value) {
-
+        if (size >= arrayList.length) {
+            resizeArrayList();
+        }
+        arrayList[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size) {
+            throw new ArrayIndexOutOfBoundsException("Index out of array size");
+        }
+        if (size == arrayList.length) {
+            resizeArrayList();
+        }
+        modifiedArrayByIndex(index);
+        arrayList[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new ArrayIndexOutOfBoundsException("Index out of array size");
+        }
```

Comment:
move this to separate method and use it in set(), get() and remove()

Reply:
done